src/controllers/objecting.entity.name.controller.ts

```diff
@@ -16,6 +22,58 @@ const validators = [
   check(DIVULGE_INFO_FIELD).not().isEmpty().withMessage(ErrorMessages.SELECT_TO_DIVULGE),
 ];
 
+const getExistingPageData = async (session: Session, res: Response, next: NextFunction) => {
+  try {
+    const objection: Objection = await getObjection(session) as Objection;
+    const existingName: string = objection.created_by.fullName;
+    const existingShareIdneity: string = (objection.created_by.shareIdentity === true) ? "yes" : "no";
+    if (existingName && existingShareIdneity) {
+      return res.render(Templates.OBJECTING_ENTITY_NAME, {
+        fullNameValue: existingName,
+        isYesChecked: existingShareIdneity === "yes",
+        isNoChecked: existingShareIdneity === "no",
+        templateName: Templates.OBJECTING_ENTITY_NAME,
+      });
+    } else {
+      return next(new Error("Existing data not present"));
+    }
+  } catch (e) {
+    logger.error(e.message);
+    return next(e);
+  }
+}
+
+const isChangeAnswerFlagPresentSetAndSetToTrue = (session: Session): boolean => {
+  const changeAnswer: boolean = retrieveFromObjectionSession(session, CHANGE_ANSWER_KEY);
+  if (changeAnswer) {
+    return changeAnswer === true;
+  }
+  return false;
+};
+
+/**
+ * GET checks for change flag and renders page
+ * @param req
+ * @param res
+ * @param next
+ */
+export const get = async (req: Request, res: Response, next: NextFunction) => {
+  const session: Session | undefined = req.session as Session;
+  if (session) {
+    if (isChangeAnswerFlagPresentSetAndSetToTrue(session)) {
+      // TODO OBJ-287 handle this more formally.
+      delete retrieveObjectionSessionFromSession(session)[CHANGE_ANSWER_KEY];
```

Comment:
I believe clearing the flag will be done on loading the check your answers page, not on the indiviudal pages

Reply:
I think we want to minimise the time it is left in the session as we don't want there to be the details to be on the page unexpectedly - the details for theis will be worked out for case OBJ-287.

Reply:
If this flag is cleared on the GET, then it won't be present on the POST to decide where to send the user next.
Don't worry about clearing the flag in this pr, 287 willl ensure the flag gets cleared when it needs to

Reply:
I think Ian is right, if we delete the change answer key here, it won't be available when 'post' is called and we'll end up going to the next page instead of back to the 'change your answers' page

Reply:
Alright removed this - it was only mean to be a stop gap - it will mean that we will have the details appearing on the page for the lifecycle of the session until 287 is completed.